src/BlockNLPModels.jl

```diff
@@ -0,0 +1,213 @@
+module BlockNLPModels
+
+using NLPModels
+using LinearAlgebra
+using SparseArrays
+
+export AbstractBlockNLPModel
+export BlockNLPModel
+export DualizedNLPblockModel
+export FullSpaceModel
+export add_block, add_links, n_tot_con, get_blockmatrix, get_matrixblocks, get_RHSvector
+
+"""
+    AbstractBlockNLPModel
+Abstract supertype for the definition of NLP problems with a block structure.
+"""
+abstract type AbstractBlockNLPModel end
+
+"""
+    AbstractLinearLinkConstraint
+Abstract supertype for the definition of linear linking constraints
+"""
+abstract type AbstractLinearLinkConstraint end
```

Comment:
Do we need to define this abstract type? If we have only one struct inheriting from this abstract type, it would not make sense.

Reply:
You're right, there is no use of this abstract type right now. But later, once we also have nonlinear linking constraints, I thought it would be useful to have this type definition to differentiate between the two types.